### PR TITLE
Slight restructure of sdk content

### LINKED ISSUE #1106: Combine Unity SDK guides into one section
The Unity SDK guides only apply to the Unity platform, but they are dispersed throughout the how-to docs. It might be more useful for Unity developers to have these docs all in one place.

### PR DESCRIPTION
**Pull request by alexandratran**
Consolidate the Unity SDK guides into the Unity section, and add the third-party libraries into the "Use SDK" section. Minor edits to related content. Fixes #1106.

Preview:
- https://docs.metamask.io/1106-sdk-reorg/wallet/how-to/use-sdk/gaming/unity/
- https://docs.metamask.io/1106-sdk-reorg/wallet/how-to/use-sdk/3rd-party-libraries/

### CODE REVIEW

File: wallet/concepts/smart-contracts.md [hunk 1]
```diff
@@ -1,13 +1,13 @@
 ---
-description: Enable your dapp to interact with smart contracts.
-sidebar_position: 10
+description: Learn about interacting with smart contracts.
+sidebar_position: 7
 ---
 
-# Interact with smart contracts
+# Smart contracts
```
Comment (joaniefromtheblock) [inline wallet/concepts/smart-contracts.md:6]: Are descriptive headings more helpful? Also the other headings seem to start with an action

File: wallet/how-to/use-sdk/gaming/unreal-engine.md [hunk 2]
```diff
@@ -6,5 +6,5 @@ sidebar_position: 2
 # Use MetaMask SDK with Unreal Engine
 
 [MetaMask SDK](../../../concepts/sdk/index.md) support for Unreal Engine games is coming soon.
```
Comment (joaniefromtheblock) [inline wallet/how-to/use-sdk/gaming/unreal-engine.md:8]: Should we link to the Unreal engine games page for the time being? https://www.unrealengine.com/en-US/solutions/games

Comment (joaniefromtheblock) [reply wallet/how-to/use-sdk/gaming/unreal-engine.md:8]: Same question for the Wagmi mention. Should that include a link as well to the official page? https://wagmi.sh

**Review by joaniefromtheblock — APPROVED** [review]
LGTM. Just had a couple of questions

File: wallet/concepts/smart-contracts.md (hunk 1 above)
Comment (alexandratran) [reply wallet/concepts/smart-contracts.md:6]: This was moved to the Concepts section – pages in that section don't start with a verb.

**Pull request merged**